AY2122S1-CS2103T-T17-3/tp
# Add Next Meeting

### Description

Add NextMeeting class which encapsulates:
* date
* startTime
* endTime
* location
of a meeting. The prefix of a NextMeeting is `m/`.

NextMeeting information is shown in PersonViewPanel as well as PersonCard.

Format to submit a NextMeeting as a string is:
`m/dd-mm-yyyy (hh:mm~hh:mm), {non-empty location string)`.

For example, a valid NextMeeting is:
`m/25-12-2021 (00:00~23:59), Santa's Workshop`

#### Breaking change (FIXED with 565c2e5):

PersonCard width behaves very differently when NextMeeting string exceeds PersonCard length or is within PersonCard length.

<img width="743" alt="Screenshot 2021-10-18 at 10 41 41 PM" src="https://user-images.githubusercontent.com/77223551/137753458-0583b6e6-efc9-443f-aa4d-40f369b0c33e.png">

### Type of change


### How to test

From the sample address book, you can run this command to check out the behaviour of NextMeeting.

Command | Behaviour
-- | --
add n/Carl e/carl@gmail.com | add a Client named Carl with 'No meeting planned'
add n/Nick e/nick@gmail.com m/20-12-2021 (12:00~12:15), Zoom | add a Client named Nick with NextMeeting set up
edit 6 m/18-10-2021 (18:00~19:00), NUS | update Carl's NextMeeting

### Checklist
